plugin.rb

```diff
@@ -18,68 +18,46 @@
 register_asset 'lib/autocomplete.js'
 
 after_initialize do
-  load File.expand_path('../lib/discourse_algolia/algolia_helper.rb', __FILE__)
-
-  module ::DiscourseAlgolia
-    PLUGIN_NAME = "discourse-algolia".freeze
-
-    class Engine < ::Rails::Engine
-      engine_name DiscourseAlgolia::PLUGIN_NAME
-      isolate_namespace DiscourseAlgolia
+  load File.expand_path("../lib/discourse_algolia.rb", __FILE__)
+
+  USER_EVENTS ||= %i{
+    user_created
+    user_updated
+    user_destroyed
+  }
+
+  POST_EVENTS ||= %i{
+    post_created
+    post_edited
+    post_destroyed
+    post_recovered
+  }
+
+  TAG_EVENTS ||= %i{
+    tag_created
+    tag_updated
+    tag_destroyed
+  }
+
+  USER_EVENTS.each do |event|
+    DiscourseEvent.on(event) do |user|
+      next unless SiteSetting.algolia_enabled?
```

Comment:
I'm not sure why we don't wrap it all with one single `SiteSetting.algolia_enabled?` ? It's not going to change mid iteration